src/libs/actions/Policy/Policy.ts

```diff
@@ -3740,6 +3749,634 @@ function getAdminPoliciesConnectedToNetSuite(): Policy[] {
     return Object.values(allPolicies ?? {}).filter<Policy>((policy): policy is Policy => !!policy && policy.role === CONST.POLICY.ROLE.ADMIN && !!policy?.connections?.netsuite);
 }
 
+/**
+ * Call the API to enable custom report title for the reports in the given policy
+ * @param policyID - id of the policy to apply the limit to
+ * @param enabled - whether custom report title for the reports is enabled in the given policy
+ */
+function enablePolicyDefaultReportTitle(policyID: string, enabled: boolean) {
+    const policy = getPolicy(policyID);
+    const previousReportTitleField = policy?.fieldList?.[CONST.POLICY.FIELD_LIST_TITLE_FIELD_ID] ?? {};
+
+    const titleFieldValues = enabled ? {} : {fieldList: {[CONST.POLICY.FIELD_LIST_TITLE_FIELD_ID]: {...previousReportTitleField, defaultValue: CONST.POLICY.DEFAULT_REPORT_NAME_PATTERN}}};
+    const titleFieldPendingValues = enabled ? {} : {fieldList: CONST.RED_BRICK_ROAD_PENDING_ACTION.UPDATE};
+
+    const optimisticData: OnyxUpdate[] = [
+        {
+            onyxMethod: Onyx.METHOD.MERGE,
+            key: `${ONYXKEYS.COLLECTION.POLICY}${policyID}`,
+            value: {
+                shouldShowCustomReportTitleOption: enabled,
+                ...titleFieldValues,
+                pendingFields: {
+                    shouldShowCustomReportTitleOption: CONST.RED_BRICK_ROAD_PENDING_ACTION.UPDATE,
+                    ...titleFieldPendingValues,
```

Comment:
When users disable "Custom report names" option, we will hide two sub-options. Thus, I don't think we need to set pendingFields.fieldList. Or could you help to explain why we need this line?